#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()

gen.add("max_iterations", int_t, 0, "maximum iterations", 1000, 0, 10000)
gen.add("correspondence_randomness", int_t, 0, "Number of nearest features to use",
                                     2, 0, 10)
gen.add("similarity_threshold", double_t, 0, "Polygonal edge length similarity threshold",
        0.9, 0.0, 1.0)
gen.add("max_correspondence_distance", double_t, 0, "inlier threshold",
        0.0075, 0.0, 1.0)
gen.add("inlier_fraction", double_t, 0, "inlier fraction",
        0.25, 0.0, 1.0)

exit (gen.generate (PACKAGE, "jsk_pcl_ros", "FeatureRegistration"))
